File: web-admin/app/constants/error-codes.ts
export const RECONCILIATION_ERROR_MESSAGES: Record<string, string> = {
  RECONCILIATION_BATCH_NOT_FOUND: "未找到对账批次，请刷新后重试",
  RECONCILIATION_DELTA_NOT_FOUND: "未找到差异记录，请确认数据是否已变更",
  RECONCILIATION_TASK_ALREADY_OPEN: "该差异已有未关闭处置任务，请直接跟进现有任务",
  RECONCILIATION_UNSUPPORTED_ACTION: "当前操作不被支持，请检查请求参数或状态",
  RECONCILIATION_INVALID_BILLING_CYCLE: "账期格式错误，请使用 YYYY-MM-DD",
}

export const resolveReconciliationErrorMessage = (code?: string, fallback = "操作失败，请稍后重试"): string => {
  if (!code) return fallback
  return RECONCILIATION_ERROR_MESSAGES[code] || fallback
}

File: backend/internal/contracts/error_codes.go
package contracts

const (
	// After-sales domain error codes.
	ErrCodeAfterSalesServiceUnavailable = "AFTER_SALES_SERVICE_UNAVAILABLE"
	ErrCodeAfterSalesCaseNotFound       = "AFTER_SALES_CASE_NOT_FOUND"
	ErrCodeAfterSalesInvalidAction      = "AFTER_SALES_INVALID_ACTION"
	ErrCodeAfterSalesInvalidState       = "AFTER_SALES_INVALID_STATE_TRANSITION"
	ErrCodeAfterSalesCoreFieldsFrozen   = "AFTER_SALES_CORE_FIELDS_FROZEN"
	ErrCodeAfterSalesReasonRequired     = "AFTER_SALES_REASON_REQUIRED"
	ErrCodeAfterSalesRefundDuplicated   = "AFTER_SALES_REFUND_DUPLICATED"
	ErrCodeAfterSalesReverseNotAllowed  = "AFTER_SALES_REVERSE_NOT_ALLOWED"
	ErrCodeAfterSalesReverseNotFound    = "AFTER_SALES_REVERSE_WAYBILL_NOT_FOUND"
	ErrCodeAfterSalesReverseMismatch    = "AFTER_SALES_REVERSE_ORDER_MISMATCH"

	// Subscription reconciliation domain error codes.
	ErrCodeReconciliationBatchNotFound       = "RECONCILIATION_BATCH_NOT_FOUND"
	ErrCodeReconciliationDeltaNotFound       = "RECONCILIATION_DELTA_NOT_FOUND"
	ErrCodeReconciliationTaskAlreadyOpen     = "RECONCILIATION_TASK_ALREADY_OPEN"
	ErrCodeReconciliationUnsupportedAction   = "RECONCILIATION_UNSUPPORTED_ACTION"
	ErrCodeReconciliationInvalidBillingCycle = "RECONCILIATION_INVALID_BILLING_CYCLE"
)
